### DISCUSSION TITLE
Stream encountered HTTP error: 403

### DISCUSSION THREAD

**Discussion by kenryu42** [tweepy/tweepy#1878] [Questions] [answered]
```
class Listener(tweepy.Stream):
    def on_status(self, status):
        print(status.user.screen_name + ": " + status.text)


stream_tweet = Listener(api_key, api_key_secret, access_token, access_token_secret)

keywords = ["2002", "#python"]

stream_tweet.sample()
```
I am getting "**Stream encountered HTTP error: 403**" when using the above code.
Other method such as **_home_timeline_** etc is working fine.

I am using tweepy 4.8.0 and python 3.9.10 on Mac OS.
I have **Elavated Access** of twitter developers api.

Is anyone experiencing the same errors or have any ideas for making the stream work?

**Comment by IndoRexian** [comment]
I was also getting the same error while using `tweepy.Streams` but it has been deprecated if your App is new. After I changed to `tweepy.StreamingClient` and it was working well.

**Reply by ItsAditya-xyz** [reply]
thanks, It works. But I am trying to find screen_name of the tweet author. But it looks like there is no way to do in tweepy.StreamingClient. Any helps will be appreciated

**Reply by Harmon758** [reply]
You can use the `referenced_tweets.id.author_id` [expansion](https://tweepy.readthedocs.io/en/v4.10.0/expansions_and_fields.html#expansions).

Please don't repeat the same unrelated question in discussions.
If you have a new question, feel free to start a new discussion.

**Comment by Harmon758** [comment]
Are you using a Twitter Developer App created on or after last Friday (2022-04-29)?

**Comment by kenryu42** [comment]
> Are you using a Twitter Developer App created on or after last Friday (2022-04-29)?

Yes! I just created the app yesterday!

**Comment by kenryu42** [comment]
> I was also getting the same error while using `tweepy.Streams` but it has been deprecated if your App is new. After I changed to `tweepy.StreamingClient` and it was working well.

Could you share some of your code? 
And are you enabling both OAuth 2.0 and OAuth 1.0a for your app?

**Comment by elon2022** [comment]
兄弟你是中国人吗，如果是一起交流一下啊。我的微信：    Cc18991001        我跟你遇到了一模一样的问题

**Comment by elon2022** [comment]
兄弟你是中国人吗？我跟你遇到一模一样的问题，我也是申请好了高级权限，然后使用此功能的，同样的是403. 加个微信啊： Cc18991001

**Comment by elon2022** [comment]
Brother, are you Chinese? I have the exact same problem as you, I also applied for advanced permissions, and then used this function, the same is 403. Add a WeChat: Cc18991001

**Comment by elon2022** [comment]
Brother, I have been working on this problem for a long time, and I haven't solved it yet. Can you add a WeChat or a telegram: @musk1992 Let's solve this problem together

**Comment by Harmon758** [comment, marked as answer]
New Apps:
> will not be able to gain access to v1.1 statuses/sample and v1.1 statuses/filter

https://twittercommunity.com/t/deprecation-announcement-removing-compliance-messages-from-statuses-filter-and-retiring-statuses-sample-from-the-twitter-api-v1-1/170500

You'll have to use Twitter API v2.

**Comment by elon2022** [comment]
I have the same problem as you: my permissions are also enabled. Then it is too outrageous to not use 403, and the version is also 4.8.0

**Comment by IndoRexian** [comment]
> > I was also getting the same error while using `tweepy.Streams` but it has been deprecated if your App is new. After I changed to `tweepy.StreamingClient` and it was working well.
> 
> Could you share some of your code? And are you enabling both OAuth 2.0 and OAuth 1.0a for your app?


I have enabled both 2.0 and 1.0a.
What I did was create a whole new app (just so that everythin goes well), Enabled both 2.0 and 1.0a with read and write perms.

Then I used `tweepy.StreamingClient` as it's of v2.0.
Here is a piece of code:
```py
auth = tweepy.OAuth1UserHandler(
    consumer_key, consumer_secret, access_token, access_token_secret
)
api = tweepy.API(auth)

class ConnectionTester(tweepy.StreamingClient):
    def on_tweet(self, tweet: tweepy.Tweet):
        id = tweet.id
        api.retweet(id)

stream = ConnectionTester(bearer_token)
stream.add_rules(
    [
        tweepy.StreamRule("#Python"),
        tweepy.StreamRule("#Tweepy"),
    ]
)
stream.filter()
```

**Reply by rushkii** [reply]
Any update about this problem? I have the exact same problem

**Reply by Harmon758** [reply]
What problem are you referring to? This answer isn't about a problem.

**Reply by ItsAditya-xyz** [reply]
Thanks, it works! how to find screen_name from tweet object?
i recall on_status method in StreamListener has status.user.screen_name attribute..

**Reply by IndoRexian** [reply]
> Thanks, it works! how to find screen_name from tweet object? i recall on_status method in StreamListener has status.user.screen_name attribute..

You can get that through `get_user()`
https://docs.tweepy.org/en/v4.10.0/api.html#tweepy.API.get_user

https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/user

**Reply by Harmon758** [reply]
If you want the username while using [`StreamingClient`](https://tweepy.readthedocs.io/en/v4.10.0/streamingclient.html), you can use the `referenced_tweets.id.author_id` [expansion](https://tweepy.readthedocs.io/en/v4.10.0/expansions_and_fields.html#expansions).
